src/test/java/uk/gov/ons/ctp/response/action/endpoint/ActionRuleEndpointIT.java

```diff
@@ -0,0 +1,202 @@
+package uk.gov.ons.ctp.response.action.endpoint;
+
+import static org.junit.Assert.assertEquals;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+import java.time.OffsetDateTime;
+import java.util.UUID;
+import javax.transaction.Transactional;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.embedded.LocalServerPort;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.rules.SpringClassRule;
+import org.springframework.test.context.junit4.rules.SpringMethodRule;
+import uk.gov.ons.ctp.common.UnirestInitialiser;
+import uk.gov.ons.ctp.common.utility.Mapzer;
+import uk.gov.ons.ctp.response.action.domain.repository.*;
+import uk.gov.ons.ctp.response.action.representation.*;
+
+@Slf4j
+@ContextConfiguration
+@ActiveProfiles("test")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class ActionRuleEndpointIT {
+
+  @Autowired private ResourceLoader resourceLoader;
+
+  @Autowired private ActionRepository actionRepository;
+
+  @Autowired private ActionCaseRepository actionCaseRepository;
+
+  @Autowired private ActionPlanRepository actionPlanRepository;
+
+  @Autowired private ActionRuleRepository actionRuleRepository;
+
+  @Autowired private ActionPlanJobRepository actionPlanJobRepository;
+
+  @ClassRule public static final SpringClassRule SPRING_CLASS_RULE = new SpringClassRule();
+
+  @Rule public final SpringMethodRule springMethodRule = new SpringMethodRule();
+
+  @LocalServerPort private int port;
+
+  @Autowired private ObjectMapper mapper;
+
+  private Mapzer mapzer;
+
+  @Before
+  @Transactional
+  public void setup() {
+    mapzer = new Mapzer(resourceLoader);
+    UnirestInitialiser.initialise(mapper);
+    actionCaseRepository.deleteAll();
+    actionRepository.deleteAll();
+    actionPlanJobRepository.deleteAll();
+    actionRuleRepository.deleteAll();
+    actionPlanRepository.deleteAll();
+  }
+
+  private ActionPlanDTO createActionPlan() throws UnirestException {
```

Comment:
Would be nicer to have the private methods below the test methods

Reply:
Fixed